drivers/note/note_driver.c

```diff
@@ -416,203 +474,346 @@ void sched_note_start(FAR struct tcb_s *tcb)
       return;
     }
 
-  /* Copy the task name (if possible) and get the length of the note */
+  for (driver = g_note_drivers; *driver; driver++)
+    {
+      if (note_start(*driver, tcb))
+        {
+          continue;
+        }
+
+      if (!formatted)
```

Comment:
let's check add isn't null before formatting in all sched_note_xxx function.

Reply:
updated, but checks in note_add don't need to be removed

Reply:
but the check is redundant now